### PR TITLE
assetsys: Add `assetsys_mount_data()` to mount data buffers

### PR DESCRIPTION
**Pull request by RobLoach**
This change attempts to tackle #53 and adds a `assetsys_mount_data()` function to mount data from an archive data buffer. This paves the path to allow disabling STDIO, and cases in which the target already has the file loaded into memory.

```
assetsys_error_t assetsys_mount_data( assetsys_t* sys, void const* data, size_t size, char const* mount_as )
```

It splits the `assetsys_mount()` into two different internal functions to save on code duplication. These functions may or may not be named correctly:
- `assetsys_internal_create_mount()`: Prepares a new mount for the assetsys instance
- `assetsys_internal_mount_files()`: Iterates through the mount system and indexes the files

Would love your thoughts on this. Eventually, I'd like to be able to define `MINIZ_NO_STDIO` to skip cases where STDIO isn't needed.

### CODE REVIEW

**Comment by mattiasgustavsson** [comment]
I think that looks great! Only thing is, I am not sure about, is the name `assetsys_mount_data` - it's just that in the context of assetsys, the term `data` is almost overused as is, in names and documentation :P I think it would be better to call the function `assetsys_mount_from_memory` in a similar naming style as stb_image.h, which has `stbi_load` and `stbi_load_from_memory` iirc. It's no big thing though, and I am not dead set on it. What do you think?

**Comment by RobLoach** [comment]
- Renamed the function to `assetsys_mount_from_memory`
- Added a test case for it
- Also tried compiling with `MINIZ_NO_STDIO`, and it seems to function properly

File: assetsys.h [hunk 1]
```diff
@@ -6355,6 +6425,57 @@ void test_assetsys( void ) {
     assetsys_destroy( assetsys );
 
     TESTFW_TEST_END();
+
+    TESTFW_TEST_BEGIN( "Test mounting data and loading file" );
+        {
+        // Zip file with a test.txt file containing "Hello, World!"
+        const char data[]  = {
```
Comment (mattiasgustavsson) [inline assetsys.h:6432]:
```suggestion
        const unsigned char data[]  = {
```
Looks like there's a build failing, I assume switching to unsigned char would fix it

**Comment by mattiasgustavsson** [comment]
Oh, I almost forgot, there's a "contributors" section near the end of the file, just before the license. Maybe add your name there (only if you want to of course). But yeah, looks good to go!

**Comment by mattiasgustavsson** [comment]
Great! Thanks!

**Pull request merged**